src/connection.ts

```diff
@@ -93,6 +91,22 @@ export interface ConnectionOpts {
    * When omitted, use a timeout of 30 seconds.
    */
   getExpiry?: (destination: string) => Date,
+  /**
+   * Callback for the consumer to perform accounting and choose to fulfill an incoming ILP Prepare,
+   * given the identifier of the connection, sequence of the packet, and amount received.
+   *
+   * If the returned Promise is resolved, the ILP Prepare will be fulfilled; if it is rejected,
+   * the ILP Prepare will be rejected. The ILP Fulfill will be immediately sent back after
+   * the Promise is resolved.
+   *
+   * Notes:
+   * - If `shouldFulfill` is provided, all connections MUST have a `connectionTag`, otherwise
+   *   the packets are rejected.
+   * - If the total amount received on the connection exceeds the max u64, the packet will be
+   *   rejected, even if the callback is resolved.
+   * - If a sender is non-compliant with STREAM, they can trigger duplicate sequence numbers.
```

Comment:
I can't think of a specific risk here, but is it possible to check this before shouldFulfill? We should do much as we can to clean up these edge cases before passing events on to the developer

Reply:
I think this requires persistence, so the developer must handle this. If a connection was established, closed, the STREAM server restarted, and then a connection was established with the same credentials, the sender could still duplicate sequence numbers.

From the developer's perspective, they should use the connectionTag and sequence together for uniqueness in their persistence system. If there's a packet with a duplicate connectionTag and sequence number, that packet should be discarded & rejected.